src/h3/api/_api_template.py

```diff
@@ -426,43 +427,78 @@ def polygon_to_cells(self, polygon, res):
          '862830957ffffff',
          '86283095fffffff'}
         """
-        mv = _cy.polygon_to_cells(polygon.outer, res, holes=polygon.holes)
 
-        return self._out_unordered(mv)
+        # todo: not sure if i want this dispatch logic here. maybe in the objects?
+        if isinstance(h3shape, H3Poly):
+            poly = h3shape
+            mv = _cy.polygon_to_cells(poly.outer, res, holes=poly.holes)
+        elif isinstance(h3shape, H3MultiPoly):
+            mpoly = h3shape
+            mv = _cy.polygons_to_cells(mpoly.polys, res)
+        else:
+            raise ValueError('Unrecognized type: ' + str(h3shape))
 
-    # def polygons_to_cells(self, polygons, res):
-    #     # todo: have to figure out how to concat memoryviews cleanly
-    #     # or some other approach
-    #     pass
+        return self._out_unordered(mv)
 
-    def cells_to_polygons(self, cells):
+    def cells_to_h3shape(self, cells, tight=True):
         """
-        Return a list of h3.Polygon objects describing the area
-        covered by a set of H3 cells.
+        Return a H3MultiPoly describing the area covered by a set of H3 cells.
 
         Parameters
         ----------
         cells : iterable of H3 cells
+        tight : bool
+            If True, return H3Poly if possible. If False, always return H3MultiPoly
```

Comment:
Is this helpful?

I can see how this can be convenient for many cases, but maybe confusing/more work to deal with for the caller if the return type may be one of two things, and it's not easy to know in advance what they're getting.

Reply:
It is easy to know in advance what they're getting if they set `tight=False` :)

I could go either way on this, but I found it ergonomic when I'm playing with data sets manually, or if I'm demonstrating how things work with smaller examples.

And it seems to be standard practice, at least with `geopandas`, to use the smaller container when possible.
For example:

```
>>> geopandas.read_file(geodatasets.get_path('geoda airbnb')).geometry
0     POLYGON ((-87.60914 41.84469, -87.60915 41.844...
1     POLYGON ((-87.59215 41.81693, -87.59231 41.816...
2     POLYGON ((-87.62880 41.80189, -87.62879 41.801...
3     POLYGON ((-87.60671 41.81681, -87.60670 41.816...
4     POLYGON ((-87.59215 41.81693, -87.59215 41.816...
                            ...                        
72    POLYGON ((-87.69646 41.70714, -87.69644 41.706...
73    POLYGON ((-87.64215 41.68508, -87.64249 41.685...
74    MULTIPOLYGON (((-87.83658 41.98640, -87.83658 ...
75    POLYGON ((-87.65456 41.99817, -87.65456 41.998...
76    POLYGON ((-87.80676 42.00084, -87.80676 42.000...